pkg/volume/gce_pd/gce_pd.go

```diff
@@ -234,16 +237,31 @@ func (plugin *gcePersistentDiskPlugin) NewDeleter(spec *volume.Spec) (volume.Del
 }
 
 func (plugin *gcePersistentDiskPlugin) newDeleterInternal(spec *volume.Spec, manager pdManager) (volume.Deleter, error) {
-	if spec.PersistentVolume != nil && spec.PersistentVolume.Spec.GCEPersistentDisk == nil {
-		return nil, fmt.Errorf("spec.PersistentVolumeSource.GCEPersistentDisk is nil")
+	zones := make(sets.String)
+	var err error
+	if spec.PersistentVolume != nil {
+		if spec.PersistentVolume.Spec.GCEPersistentDisk == nil {
```

Comment:
Isn't there a GetVolumeSource or something that we can use?

Reply:
getVolumeSource will also return the volume spec if it's an inline volume, but here we have to make sure it's a PV